### DISCUSSION TITLE
Song lookup

### DISCUSSION THREAD

**Discussion by braidynbritt** [JohnnyCrazy/SpotifyAPI-NET#854] [Q&A]
I am currently programming a smart mirror. I want to be able to have the user say what song they want to play. How do I get the track ID from the song that they request instead of the other way around? (I have to use SpotifyAPI-NET 5.1.1, my visual studio will not let me upgrade my .NET to higher than 4.7 for some reason.)

**Comment by JohnnyCrazy** [comment]
You will have to use the search endpoint and use the best matching song or let the user select.

```csharp
var spotify = new SpotifyClient("YourAccessToken");

const results = await spotift.Search.Item(SearchRequest.Types.Track, "Song Name")
// iterate through results.Tracks.Items to use best matching song
```
